raiden/tests/unit/storage/migrations/test_v19_to_v20.py

```diff
@@ -0,0 +1,127 @@
+from raiden.storage.serialize import JSONSerializer
+from raiden.storage.sqlite import SerializedSQLiteStorage, SQLiteStorage
+from raiden.tests.utils.factories import make_32bytes, make_address
+from raiden.tests.utils.mocks import MockRaidenService
+from raiden.utils.serialization import serialize_bytes
+from raiden.utils.upgrades import UpgradeManager
+
+
+def setup_storage(db_path):
+    storage = SQLiteStorage(str(db_path))
+
+    state_changes_file = Path(__file__).parent / 'data/v19_statechanges.json'
+    state_changes_data = json.loads(state_changes_file.read_text())
+    for state_change_record in state_changes_data:
+        storage.write_state_change(
+            state_change=json.dumps(state_change_record[1]),
+            log_time=datetime.utcnow().isoformat(timespec='milliseconds'),
+        )
+
+    # Add the v18 events to the DB
+    events_file = Path(__file__).parent / 'data/v19_events.json'
+    events_data = json.loads(events_file.read_text())
+    event_tuples = []
+    for event in events_data:
+        state_change_identifier = event[1]
+        event_data = json.dumps(event[2])
+        log_time = datetime.utcnow().isoformat(timespec='milliseconds')
+        event_tuples.append((
+            None,
+            state_change_identifier,
+            log_time,
+            event_data,
+        ))
+    storage.write_events(event_tuples)
+
+    chain_state_data = Path(__file__).parent / 'data/v19_chainstate.json'
+    chain_state = chain_state_data.read_text()
+    cursor = storage.conn.cursor()
+    cursor.execute(
+        """
+        INSERT INTO state_snapshot(identifier, statechange_id, data)
+        VALUES(1, 1, ?)
+        """, (chain_state,),
+    )
+    storage.conn.commit()
+    return storage
+
+
+def test_upgrade_v19_to_v20(tmp_path):
+    old_db_filename = tmp_path / Path('v19_log.db')
+    with patch('raiden.utils.upgrades.older_db_file') as older_db_file:
+        older_db_file.return_value = str(old_db_filename)
+        storage = setup_storage(str(old_db_filename))
+        with patch('raiden.storage.sqlite.RAIDEN_DB_VERSION', new=19):
+            storage.update_version()
+        storage.conn.close()
+
+    raiden_service_mock = MockRaidenService()
+
+    our_onchain_locksroot = serialize_bytes(make_32bytes())
+    partner_onchain_locksroot = serialize_bytes(make_32bytes())
+
+    details = Mock()
+    details.our_details.address = make_address()
+    details.our_details.locksroot = our_onchain_locksroot
+    details.partner_details.address = make_address()
+    details.partner_details.locksroot = partner_onchain_locksroot
+
+    payment_channel = Mock()
+    payment_channel.token_network.detail_participants.return_value = details
+
+    payment_channel_func = Mock()
+    payment_channel_func.return_value = payment_channel
+
+    raiden_service_mock.chain.payment_channel = payment_channel_func
+
+    db_path = tmp_path / Path('v20_log.db')
```

Comment:
And what are you doing here? Upgrading v19 to v20? If so how come the db_path is `v20_log.db`?

Reply:
It should be... we always pass the "current version's DB filename" to the upgrade manager. In this case this is v20_log.db